### PR TITLE
Skip preset role creation on dashboard tenant

### PR DESCRIPTION
**Pull request by logand22**
This PR removes the default creation of preset roles and users on startup when a dashboard tenant is present. This PR is meant for Dashboard tenants where the default roles are removed via a script afterwards in favor of the DashboardAdmin roles instead. 

I don't know if this is safe in practice and would love feedback or guidance.

### CODE REVIEW

**Review by mcbattirola — APPROVED** [review]

File: lib/auth/init.go [hunk 1]
```diff
@@ -502,18 +503,23 @@ func initCluster(ctx context.Context, cfg InitConfig, asrv *Server) error {
 	}
 	span.AddEvent("completed migration legacy resources")
 
-	span.AddEvent("creating preset roles")
 	// Create presets - convenience and example resources.
-	if err := createPresetRoles(ctx, asrv); err != nil {
-		return trace.Wrap(err)
-	}
-	span.AddEvent("completed creating preset roles")
+	isDashboard := !modules.GetModules().Features().Cloud && modules.GetModules().Features().RecoveryCodes
```
Comment (zmb3) [inline lib/auth/init.go:507]: Can you use `services.IsDashboard` here? (from lib/services/license.go)

File: lib/auth/init.go [hunk 2]
```diff
@@ -502,18 +503,23 @@ func initCluster(ctx context.Context, cfg InitConfig, asrv *Server) error {
...
+	if !isDashboard {
+		span.AddEvent("creating preset roles")
+		if err := createPresetRoles(ctx, asrv); err != nil {
+			return trace.Wrap(err)
+		}
+		span.AddEvent("completed creating preset roles")
```
Comment (zmb3) [inline lib/auth/init.go:513]: The span will never complete if there's an error. Can we record the error case to the span as well?

File: lib/auth/init.go [hunk 3]
```diff
@@ -502,18 +503,23 @@ func initCluster(ctx context.Context, cfg InitConfig, asrv *Server) error {
...
-	span.AddEvent("creating preset users")
-	if err := createPresetUsers(ctx, asrv); err != nil {
-		return trace.Wrap(err)
+		span.AddEvent("creating preset users")
+		if err := createPresetUsers(ctx, asrv); err != nil {
+			return trace.Wrap(err)
+		}
+		span.AddEvent("completed creating preset users")
+	} else {
+		span.AddEvent("skipping preset role and user creation")
```
Comment (zmb3) [inline lib/auth/init.go:521]: I would just log this. You don't need to put it in the span.

File: lib/auth/init.go (hunk 1 above)
Comment (logand22) [reply lib/auth/init.go]: Using `services.IsDashboard` now in [1054acc](https://github.com/gravitational/teleport/pull/37610/commits/1054acc35e06ef535174f1eb7923c57d20805686).

File: lib/auth/init.go (hunk 3 above)
Comment (logand22) [reply lib/auth/init.go]: Switched to logging this here: [1054acc](https://github.com/gravitational/teleport/pull/37610/commits/1054acc35e06ef535174f1eb7923c57d20805686).

File: lib/auth/init.go (hunk 2 above)
Comment (logand22) [reply lib/auth/init.go:516]: Attempted to address this here: [1054acc](https://github.com/gravitational/teleport/pull/37610/commits/1054acc35e06ef535174f1eb7923c57d20805686). 

For this I wasn't too sure if the way I addressed it is okay? I see in other cases that error is recorded but through a defer function instead.

Comment (zmb3) [reply lib/auth/init.go:516]: @rosstimothy can you advise?

**Comment by logand22** [comment]
@zmb3 @mcbattirola Also noticed that passwordless option is missing when testing this locally? Are the default preset roles required for passwordless to work? In cloud we only delete `access, editor, auditor`. Perhaps I should only conditionally disable those instead of all presets?

**Comment by zmb3** [comment]
> Are the default preset roles required for passwordless to work?

@logand22 no, that shouldn't be the case.

File: lib/auth/init.go (hunk 2 above)
Comment (logand22) [reply lib/auth/init.go:512]: I reverted the changes here: [f88aaa0](https://github.com/gravitational/teleport/pull/37610/commits/f88aaa0386368f5e5a73a69c35ac86a458c7c334).

**Comment by logand22** [comment]
It was just an issue with my local environment not setting `cap` correctly. Confirmed it's unrelated to this change.

File: tool/teleport/common/teleport.go [hunk 4]
```diff
@@ -69,9 +69,6 @@ type Options struct {
 	// InitOnly when set to true, initializes config and aux
 	// endpoints but does not start the process
 	InitOnly bool
-	// EnableCloudAWSCredCmd enables hidden cloud aws cred command when true.
-	// This command does nothing in OSS.
-	EnableCloudAWSCredCmd bool
```
[the remark this answers is not in the document]
Comment (logand22) [reply tool/teleport/common/teleport.go]: rebased to remove that commit in favor of a separate PR.

**Review by rosstimothy — APPROVED** [review]

**Review by justinas — APPROVED** [review]

**Pull request merged**